torch/utils/cpp_extension.py

```diff
@@ -416,7 +416,8 @@ def load(name,
          extra_ldflags=None,
          extra_include_paths=None,
          build_directory=None,
-         verbose=False):
+         verbose=False,
```

Comment:
I actually think no changes to this file are necessary. The way I presented users for dealing with cuda is to write all CUDA code in `.cu` files, and then have a `.cpp` file which calls into the CUDA file and does the binding (`PYBIND11`) logic (see https://github.com/pytorch/extension-cpp/blob/master/cuda/lltm_cuda.cpp). This makes things like this a lot easier. So I believe this file can stay entirely unchanged, and it would be better to follow the separation pattern unless you or @fmassa think this is totally not feasible. See e.g. https://github.com/pytorch/pytorch/blob/master/test/cpp_extensions/cuda_extension.cpp for examples of this

Reply:
Those are examples for nvcc kernels though. This kind of CuDNN usage is just host code.

You can't link to cuda libraries without also running a cuda source through nvcc. If there is a situation where you just want to link to cuda libraries but you don't happen to be building a kernel, you would need this flag.

Reply:
Ok, I see. In that case, let's keep your changes here, but:
1. as @ezyang suggested `with_cuda` should default to `None` (your `with_cuda = ... or with_cuda` will still work),
2. The documentation should be something like "Determines whether CUDA headers and libraries are added to the build. If set to `None` (default), this value is automatically determined based on the existence of `.cu` or `.cuh` in `sources`. Set it to `True` to force CUDA headers and libraries to be included."

Reply:
What about for `False` but you have .cu sources? Ignore the option or try to compile cu as usual but compile cpp without cuda? Seems like that situation should at least be a warning.

Actually, does it make sense for `False` and cuda sources to be a runtime error? That might be the safest.

Reply:
Hmm maybe we don't need to do anything. E.g. `clang` can compile cuda, so maybe someone would want to compile `.cu` files with `CXX=clang` and thus set `with_cuda=False`. If not it's probably a pretty obvious compiler error

Reply:
so with_cuda=False would mean cpp_sources = cpp_sources+cuda_sources?